### ISSUE TITLE
Add infix function for ref assignment

### ISSUE THREAD

**Issue by mlutze** [flix/flix#8377]
The new ref assignment syntax is less readable than before.

```diff
- board := set_(posCoords(info0#pos), None, deref board);
+ Ref.put(set_(posCoords(info0#pos), None, Ref.get(board)), board);
```

I propose we add an infix function to `Ref` which gives us something similar to the `:=` syntax.

**Comment by JonathanStarup** [comment]
like ```x `put` thing```?

**Comment by mlutze** [comment]
> like ```x `put` thing```?

More like `x <-- thing`

**Comment by JonathanStarup** [comment]
A `Ref` is a struct now-ish, so I guess `ref.value = thing` would work

**Comment by mlutze** [comment]
> A `Ref` is a struct now-ish, so I guess `ref.value = thing` would work

Is that part of the interface or is it an implementation detail?

**Comment by JonathanStarup** [comment]
yeah good question. I think for ref its just upsides to expose it

**Comment by JonathanStarup** [comment]
When structs exists, usage of `Ref` is probably a lot rarer

**Comment by magnus-madsen** [comment]
I think - over time - refs will become rare.

**Comment by magnus-madsen** [comment]
We have Indexable and IndexableMut-- but where does that leave this ticket?

**Comment by mlutze** [comment]
Ref does not implement these since there is no index.

**Comment by magnus-madsen** [comment]
Yes, so what do you want to do here? Does it matter now that we have structs?

**Comment by JonathanStarup** [comment]
I do agree that Ref do feel really quite bad to use after the change, but not sure what do about it

**Comment by mlutze** [comment]
Simplest option is just to add some binary op to the standard library.

**Comment by magnus-madsen** [comment]
> Simplest option is just to add some binary op to the standard library.

And then? We don't have reference types anymore. Would it be a trait?

**Comment by mlutze** [comment]
> > Simplest option is just to add some binary op to the standard library.
> 
> And then? We don't have reference types anymore. Would it be a trait?

I thought reference types are a wrapper around a struct? So we just define it on the wrapper

**Comment by magnus-madsen** [comment]
What's the signature of the trait.

**Comment by mlutze** [comment]
> What's the signature of the trait.

No trait. Just a function.
